tokio-epoll-uring/src/system/lifecycle.rs

```diff
@@ -50,10 +50,69 @@ impl System {
+                    // For example, we rely on Arc<Mutex<>> to protect the `SubmitSideInner`.
+                    // A child process would have its cloned `Arc<Mutex<>>` but operate on the same MAP_SHARED memory mapping.
+                    // Disaster would ensure.
+                    .dontfork()
+                    .build(RING_SIZE)
+                    .unwrap(),
+            );
+            let flags_set_by_kernel = nix::fcntl::FdFlag::from_bits_truncate(
+                nix::fcntl::fcntl(uring.as_raw_fd(), nix::fcntl::FcntlArg::F_GETFD).unwrap(),
+            );
+            assert!(flags_set_by_kernel.contains(nix::fcntl::FdFlag::FD_CLOEXEC),
+                // NB: the kernel does this https://sourcegraph.com/github.com/torvalds/linux@v5.10/-/blob/fs/io_uring.c?L9195-9201
+                "for an individual io_uring instance, tokio-epoll-uring assumes explusive ownership \
+                 of the user-space part of the io_uring user/kernel interface; sharing it with another \
+                 process is unsafe; actually we want to prevent fork'ing, but, there's not flag for that, \
+                 so at least guard somewhat against the most common use of forking, which is fork+exec.");
+
+            // Ensure the kernel's io_uring features are sufficient for tokio-epoll-uring / it's use case in Neon Pageserver.
+            //
+            // Kernel 5.10 (Debian bullseye)      https://sourcegraph.com/github.com/torvalds/linux@v5.10/-/blob/fs/io_uring.c?L9368-9371
+            // =>      features     SINGLE_MMAP | NODROP | SUBMIT_STABLE | RW_CUR_POS | CUR_PERSONALITY | FAST_POLL | POLL_32BITS;
+            // Kernel 6.1 (Debian bookworm)       https://sourcegraph.com/github.com/torvalds/linux@v6.1/-/blob/io_uring/io_uring.c?L3534:16-3534:39
+            // => adds features     SQPOLL_NONFIXED | EXT_ARG | NATIVE_WORKERS | RSRC_TAGS | CQE_SKIP | LINKED_FILE;
+            // Kernel 6.7 (no Debian release yet) https://sourcegraph.com/github.com/torvalds/linux@v6.7/-/blob/io_uring/io_uring.c?L4054-4060
+            // => adds features     REG_REG_RING;
+            //
+            // Best man page rendering for features on the web: https://man.archlinux.org/man/io_uring_setup.2.en
+            //
+            // NODROP:
+            //  - https://github.com/golang/go/issues/31908#issuecomment-571651387
+            //  - also https://kernel.dk/io_uring-whatsnew.pdf
+            assert!(uring.params().is_feature_nodrop(), "tokio-epoll-uring has len(slots) == len(sq) == len(cq), so, in theory we don't care, but, better not worry");
+            // SUBMIT_STABLE: https://man.archlinux.org/man/io_uring_submit.3.en#NOTES ;
+            // TODO need to check each op whether it needs this, then safeguard in future ops
+            assert!(uring.params().is_feature_submit_stable(), "we can always put things into resources to avoid depending on this feature, but we haven't thought about it so far, so, let's require it");
```

Comment:
I do not understand how this could affect us, but asserting availability is probably better than omitting.